h2o-r/h2o-package/R/frame.R

```diff
@@ -3471,7 +3527,16 @@ as.data.frame.H2OFrame <- function(x, ...) {
 #' print(mins)
 #' }
 #' @export
-as.matrix.H2OFrame <- function(x, ...) as.matrix(as.data.frame.H2OFrame(x, ...))
+as.matrix.H2OFrame <- function(x, ...) {
+  .fetch.data(x,1L)   
+  nCol <- attr(x, "ncol")
+  nRow <- attr(x, "nrow")
+
+  if(nCol * nRow > .Machine$integer.max){
+    stop("It is not possible convert H2OFrame to data.frame/data.table. The H2OFrame is bigger than vector size limit for R.")
```

Comment:
Isn't this comment misleading? The user is attempting to convert to a `matrix`.

I would also suggest what to do instead -> convert to data.frame which is not limited in the same way.

Reply:
Yes, that is absolutely true.